airflow/models.py

```diff
@@ -1367,10 +1367,12 @@ def append_only_new(self, l, item):
             l.append(item)
 
     def _set_relatives(self, task_or_task_list, upstream=False):
-        if isinstance(task_or_task_list, BaseOperator):
-            task_or_task_list = [task_or_task_list]
-        for task in task_or_task_list:
-            if not isinstance(task_or_task_list, list):
+        try:
+            task_list = list(task_or_task_list)
```

Comment:
it's not super explicit here that list is used as a way to cast an iterable. btw:
```
>>> list('abc')
['a', 'b', 'c']
```

what about, instead of the try block:
```
if not isinstance(task_list, (tuple, list):
    task_list = [task_list]
```